### ISSUE TITLE
plume_simulation: port lespp_fvm → new les_fvm_abl sub-package (L3 resolved-flow LES)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/research_notebook#17]
## Context

Port the more advanced LES snippet at [`jej_vc_snippets/methane_retrieval/lespp_fvm_jax.py`](https://github.com/jejjohnson/jej_vc_snippets/blob/main/methane_retrieval/lespp_fvm_jax.py) (860 lines) into `projects/plume_simulation/` as a new `les_fvm_abl` sub-package, sibling to the existing `les_fvm` (L2 Eulerian dispersion, prescribed wind). This is the **L3 fidelity** tier the `les_fvm` README already flags as a future addition: full resolved-flow LES with a solved wind field + thermodynamics + Smagorinsky SGS.

## Honest scoping

The snippet is a **skeleton**, not a working solver. It defines the state container, BC primitives, sponge damping, ERA5 profiles, surface heat flux, and 2-D Gaussian CH₄ source — but the actual time integration, Smagorinsky SGS, momentum RHS, and pressure projection are explicitly marked as *"simplified for space"* (lines 813, 824).

So this is a **build**, not a verbatim port. We will:
1. Keep the snippet's architecture, BCs, and ABL thermodynamics.
2. Fill in the missing pieces: momentum RHS, Smagorinsky, fractional-step pressure projection, proper time integration.
3. Use `finitevolX` + `spectraldiffx` for the C-grid container, WENO5 advection, 3-D Helmholtz/Poisson solver, and diagnostics.
4. Adopt the finitevolX C-grid convention (U/V horizontally staggered, W on `[Nz+1, Ny, Nx]` interfaces) — **not** the snippet's collocated grid — to stay consistent with the existing `les_fvm` sibling.

## Math summary

Governing equations (Boussinesq, potential-temperature formulation, passive tracer):

```
∂_t u_i + ∂_j(u_j u_i)    = -(1/ρ₀) ∂_i p + ∂_j[(ν + ν_t)(∂_j u_i + ∂_i u_j)] + δ_{i3} b − α_sp(z)(u_i − u_{i,targ})
∂_t θ' + u_j ∂_j θ'       = −w ∂_z θ₀ + ∂_j[(κ + κ_t) ∂_j θ'] + Q_θ,sfc
∂_t C + ∂_j(u_j C)        = ∂_j[(κ_c + κ_{t,c}) ∂_j C] + S_CH4(x, y, t)
∂_i u_i                   = 0
```

with:
- **Buoyancy**: `b = g · θ' / θ_ref`
- **Smagorinsky SGS**: `ν_t = (C_s Δ)² |S|`, `|S| = √(2 S_ij S_ij)`, `Δ = (dx·dy·dz)^(1/3)`; turbulent Prandtl/Schmidt `Pr_t = Sc_t = 0.7 ⇒ κ_t = ν_t / Pr_t`.
- **Background profiles** (ERA5-like): log wind + well-mixed CBL + capping inversion (Stull 1988).
- **Sponge**: `α_sp(z) = α_max · sin²(π ζ / 2)` for `ζ ∈ [0, 1]` into the top 25%.
- **Surface heat flux**: `∂_t θ'|_{z=0} += H₀ / dz`.
- **Surface CH₄ source**: 2-D Gaussian in `(x, y)` at `k=0`, `σ = 3 · max(dx, dy)`.

### Numerics — fractional-step (Chorin) projection

1. Predict: `u_i* = u_i^n + Δt · RHS_i^n` (advection + diffusion + buoyancy + sponge, no pressure).
2. Solve Poisson: `∇²p = (ρ₀/Δt) ∇·u*` on periodic(x) × periodic(y) × Neumann(z).
3. Correct: `u_i^(n+1) = u_i* − (Δt/ρ₀) ∂_i p`.

Spatial discretisation:
- Momentum advection: flux-form on C-grid (finitevolX `Advection3D` horizontal WENO5 + our `_vertical_ops` for vertical).
- Scalar (θ, C) advection: same.
- SGS/molecular diffusion: central differences on faces (finitevolX `Diffusion3D` + vertical central).
- Pressure Poisson: `spectraldiffx.SpectralHelmholtzSolver3D` with `(fft, fft, dct)` mixed BCs (λ=0).

Time-step constraints (monitored, not auto-enforced):
- Advective CFL: `Δt · max(|u|/dx, |v|/dy, |w|/dz) ≲ 0.5`
- Diffusive: `Δt · max(ν + ν_t) / min(dx, dy, dz)² ≲ 0.25`

Time integrator: explicit SSPRK3 from `finitevolx._src.timestepping.explicit_rk`, with projection applied after each RK substep.

## Package layout

New sibling sub-package `les_fvm_abl/` — leaves `les_fvm` untouched (no regression risk).

```
projects/plume_simulation/src/plume_simulation/les_fvm_abl/
├── __init__.py               # public re-exports + __all__
├── state.py                  # AtmosphericState (u, v, w, θ', C, t) PyTree
├── grid.py                   # ABLGrid3D wrapper (thin extension of les_fvm.PlumeGrid3D)
├── profiles.py               # create_era5_profiles (log wind + CBL + inversion)
├── sponge.py                 # compute_sponge_damping + Rayleigh-damping tendency
├── boundary.py               # no-slip / free-slip / no-penetration + periodic lateral
├── sgs.py                    # Smagorinsky ν_t, turbulent Prandtl/Schmidt closure
├── momentum.py               # momentum advection + SGS diffusion + buoyancy
├── scalar.py                 # θ' and C advection + SGS diffusion + sources
├── projection.py             # fractional-step Poisson solve (spectraldiffx)
├── source.py                 # 2-D surface Gaussian CH₄ source
├── surface.py                # surface heat flux forcing
├── dynamics.py               # full RHS: assembles momentum + scalar + forcings
├── simulate.py               # two-phase runner (spin-up → plume) → xarray
└── _vertical_ops.py          # upwind z-flux + central z-Laplacian (reuse pattern)
```

Estimated **900–1200 lines** new code, ~1.5–2× the size of `les_fvm`, because momentum + projection + SGS are new surface area. The snippet maps 1-to-1 onto `state.py`, `profiles.py`, `sponge.py`, `boundary.py`, `source.py`, `surface.py`; everything else is new work.

## Supporting material

**Notebooks** (parallel to existing `les_fvm/` notebooks):
- `notebooks/les_fvm_abl/00_abl_les_derivation.md` — Boussinesq derivation, Smagorinsky closure, fractional-step method, ABL BCs, ERA5 initialization, spin-up heuristics. References Stull 1988, Sullivan & Patton 2011, Moeng 1984, Chorin 1968.
- `notebooks/les_fvm_abl/01_convective_bl_spinup.ipynb` — demo: 1-hour spin-up with positive surface heat flux; show TKE growth, `⟨w'²⟩(z)`, boundary-layer depth diagnosis.
- `notebooks/les_fvm_abl/02_methane_plume.ipynb` — 2-hour spin-up + 1-hour source; concentration iso-surfaces, column integral, plume meander.
- `notebooks/les_fvm_abl/03_les_vs_puff.ipynb` — comparison against `gauss_puff` at matched Q, U, stability: peak-normalized centreline, log-scale pointwise, shape ratio. Expected to disagree more than L2 because LES resolves turbulent meander.

**Tests** (`tests/les_fvm_abl/`, ~180 tests target):
- `test_profiles.py` — log-wind values, well-mixed θ, inversion slope, z₀ safety.
- `test_sponge.py` — α=0 below sponge, sin² shape, α=α_max at top.
- `test_boundary.py` — no-slip zeroes u,v at k=0; free-slip zeroes du/dz at k=Nz-1; no-penetration zeroes w at k=0, k=Nz; periodic lateral round-trip.
- `test_sgs.py` — ν_t = 0 for uniform flow; ν_t ∝ |S| on solid-body rotation; κ_t = ν_t / Pr_t.
- `test_projection.py` — after projection, ∇·u ≤ tol; Helmholtz solver sign correct; zero-velocity input → zero correction.
- `test_momentum.py` — advection conserves momentum integral with no-slip walls; rest state + stratification → no spurious motion.
- `test_scalar.py` — passive scalar mass conservation with periodic + no-flux top/bottom + source = Q·Δt; buoyancy coupling sign.
- `test_source.py` — normalized Gaussian sums to 1; periodic-wrap distance; k=0 only.
- `test_surface.py` — flux raises bottom-layer θ by H₀·Δt/dz.
- `test_simulate.py` — short spin-up returns finite values; divergence < tol; mass budget; regression on small 16³ run.

**Dependencies**: no new ones — `finitevolX>=0.0.41` already re-exports `SpectralHelmholtzSolver3D` via `spectraldiffx`. Just extend the `execute-plume-simulation` pixi-task glob to cover `les_fvm_abl` notebooks.

**`myst.yml`**: add a new subsection "Resolved-flow LES (L3)" with the four notebooks.

**`README.md`**: add `les_fvm_abl` bullet to the Sub-packages list and a Public API block.

## Public API

```python
from plume_simulation.les_fvm_abl import (
    # state
    AtmosphericState,
    # grid + init
    ABLGrid3D, make_abl_grid,
    create_era5_profiles,
    # physics primitives (for tinkering)
    compute_sponge_damping, smagorinsky_nu_t,
    apply_no_slip_bottom, apply_free_slip_top, apply_no_penetration,
    momentum_tendency, scalar_tendency,
    pressure_projection,
    surface_heat_flux_tendency, surface_methane_source_tendency,
    # high-level runner
    simulate_abl_methane_plume,
)
```

Demo one-liner (mirrors the existing `simulate_eulerian_dispersion`):

```python
ds = simulate_abl_methane_plume(
    domain=(2000.0, 2000.0, 1500.0),   # Lx, Ly, H [m]
    resolution=(64, 64, 32),           # Nx, Ny, Nz  (coarse for demo)
    dt=0.5,
    spin_up_time=1800.0,               # 30 min coarse-grid spin-up
    plume_time=1800.0,
    geostrophic_wind=(5.0, 0.0),
    theta_surface=288.0, zi=800.0,
    surface_heat_flux=0.05,            # K·m/s → convective BL
    emission_rate=0.1,                 # kg/s CH4
    source_location=(1000.0, 1000.0),
    save_interval=60.0,
    seed=42,
)
# → xarray.Dataset with u, v, w on C-grid, theta_perturbation, concentration
```

## Phasing — reviewable checkpoints

Each phase is self-contained and testable. A stall in phase B does not block A from merging.

| Phase | Deliverable | Rough size | PR |
|-------|-------------|------------|----|
| **A** | Grid + state + profiles + sponge + BCs + SGS + tests for those. No dynamics yet. | ~400 lines + ~60 tests | PR 1 |
| **B** | Momentum RHS + pressure projection + tests; verify `∇·u ≤ 1e-10` on forced flow. | ~300 lines + ~40 tests | PR 2 |
| **C** | Scalar (θ, C) RHS + surface sources + full `dynamics.py` + `simulate.py` + smoke-test run. | ~300 lines + ~40 tests | PR 3 |
| **D** | Notebooks (00/01/02/03) + README + `myst.yml` + docs polish. | docs-only | PR 4 |

## Key risks + mitigations

- **Coarse grids oscillate**: LES at 32³ is small enough that Smagorinsky over-damps. Default demo: 64³. Document that production needs 128³+. Keep 16³ "smoke test" for fast tests.
- **Pressure projection cost**: `SpectralHelmholtzSolver3D` is FFT-based — O(N log N) per step. Should be <0.1 s/step at 64³; confirm in phase B.
- **Spin-up expense**: 2-hour spin-up at dt=0.5 s = 14,400 steps. Demo notebook will use a drastically shortened 30-min spin-up on a coarse grid, plus an opt-in "full-fidelity" flag.
- **Sub-package naming**: `les_fvm_abl` is descriptive but verbose; alternatives — `les_fvm_l3`, `les_abl`, `les_resolved`. Default to `les_fvm_abl`; open to changing.

## References

1. Stull (1988) — *An Introduction to Boundary Layer Meteorology*
2. Sullivan & Patton (2011) — ABL LES
3. Canepa et al. (2022) — Methane plume LES in ABL
4. Davies (1976, 1983) — Sponge layer boundary treatment
5. Yver et al. (2013) — GML NOAA methane plume modeling
6. Moeng (1984) — pseudo-spectral LES
7. Chorin (1968) — fractional-step projection
